### DISCUSSION TITLE
Console de pilotage

### DISCUSSION THREAD

**Discussion by ChrisThe01** [MTESSDev/FRW#227] [Annonces]
Bonjour,

Nous avons travaillé avec Dany et Vincent pour déterminer la façon de trapper les erreurs de notre traitement dans le console de pilotage. Nous étions conseillé d'inscrire dans le ficher "default.v0.transmission.yml" les lignes suivantes :
check_response:
        throw_exception_if_body_not_contains_all:
          - '"CodRetour":0' # Code de retour ou un mot retourné par votre api afin de valider que tout est concluant
Lorsque le traitement _**fonctionne**_, nous avons dans la console de pilotage le suivant :
![image](https://github.com/MTESSDev/FRW/assets/129303869/0e9747ca-c111-4acc-8693-9355426dbfa3)
N.B.:
1) Lorsqu'il tombe en erreur la première, il tombe en "REPRISE". Après quelques temps, le statu change pour SUSPENDU.
2) Il n'y a rien dans l'onglet PARAMÈTRE et nous avons rien qui nous indique le nom du travail. De plus, le console dit qu'il est en erreur.
Par contre, lorsque le traitement **_ne fonctionne pas_**, nous avons le suivant : 
![image](https://github.com/MTESSDev/FRW/assets/129303869/ac1ddaa5-2f7b-47ec-a597-51f1e4d93781)
Bizarre !!

**Comment by ChrisThe01** [comment]
Y-a-t-il quelqu'un qui aider à résoudre ce problème ?

**Reply by anisite** [reply]
Bonjour, il y a beaucoup de choses en même temps dans votre message, en premier lieu, nous allons monter un correctif pour ajouter du détail plus facile à rechercher pour vous aider à retrouver vos messages, sous peu, en deuxième, je regarde dans votre config et check_response n'est pas à la bonne place dans votre fichier, encore, donc le check response n'est pas actif, ce qui explique que le message est traité en "FIN" pour l'autre cas ou ça se reprend une fois dans la console, c'est by design, le traitement se réessaie une fois, au cas ou votre système serait enfin fonctionnel, et suspend la demande en attendant que quelqu'un fasse quelque chose.

N'oubliez pas une chose : c'est votre formulaire qui doit valider l'intégrités des données et non pas le backend qui reçoit l'information. Notre file de message FRW n'est pas faite pour gérer vos validations fonctionnelles, le message devrait en tout temps être accepté par votre backend (excepté bien sûr les cas ou quelque chose d'inattendu arrive vraiment, ex: un valeur nulle imprévue ou quelque chose qui n'aurait jamais du passer)

Donc SVP corrigez votre ``default.v0.transmission.yml`` et regraduez le, tout sera ok ensuite

**Comment by ChrisThe01** [comment]
Merci, ça fonctionne maintenant. Bonne journée,
